presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -196,7 +196,7 @@ private PlanBuilder appendInPredicateApplyNode(PlanBuilder subPlan, InPredicate
 
         subPlan.getTranslations().put(inPredicate, inPredicateSubquerySymbol);
 
-        return appendApplyNode(subPlan, inPredicate, subqueryPlan.getRoot(), Assignments.of(inPredicateSubquerySymbol, inPredicateSubqueryExpression), correlationAllowed);
+        return appendApplyNode(subPlan, inPredicate, subqueryPlan.getRoot(), Assignments.of(inPredicateSubquerySymbol, castToRowExpression(inPredicateSubqueryExpression)), correlationAllowed);
```

Comment:
Usually we do the cast at the very last step when replacing expressions. This will force us to build all the necessary infra for RowExpression then start a one-time migration. It would be cleaner and easier to revert.